### PR TITLE
Refs [TASK][35604] Handle event Setting Screen

### PR DESCRIPTION
**Pull request by zGararz**
## Related Tickets
- [#Task 35604: Handle event Setting Screen](https://edu-redmine.sun-asterisk.vn/issues/35604)
## WHAT
- 
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://github.com/framgia/coding-standards/tree/master/eng/android |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/example/itbook/ui/settings/SettingsFragment.kt [hunk 1]
```diff
@@ -1,17 +1,60 @@
...
 
+import android.widget.CompoundButton
 import com.example.itbook.R
 import com.example.itbook.base.BaseFragment
+import com.example.itbook.utils.MySharedPreferences
+import com.example.itbook.utils.restart
+import com.example.itbook.utils.setLanguage
+import kotlinx.android.synthetic.main.fragment_settings.*
 
+@Suppress("DEPRECATION")
```
Comment (datdt-1935) [inline app/src/main/java/com/example/itbook/ui/settings/SettingsFragment.kt]: cái gì bị deprecated thế em?

File: app/src/main/java/com/example/itbook/ui/settings/SettingsFragment.kt [hunk 2]
```diff
@@ -1,17 +1,60 @@
...
+                }
+                restart()
+            }
+        }
+    }
+
+    companion object {
+        const val STRING_LANGUAGE = "language"
+        const val DEFAULT_LANGUAGE = "en"
+        const val LANGUAGE_VI = "vi"
```
Comment (datdt-1935) [inline app/src/main/java/com/example/itbook/ui/settings/SettingsFragment.kt]: cái này để ra `strings.xml` nhé

File: app/src/main/java/com/example/itbook/utils/ActivityExtension.kt [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import java.util.*
+
+fun Activity.setLanguage(locale: String) {
+    val locale = Locale(locale)
+    val configuration = Configuration()
+    configuration.locale = locale
+    baseContext.resources.updateConfiguration(configuration, baseContext.resources.displayMetrics)
+}
+
+fun Activity.restart() {
```
Comment (datdt-1935) [inline app/src/main/java/com/example/itbook/utils/ActivityExtension.kt:18]: cái này có tác dụng gì vậy em?

File: app/src/main/java/com/example/itbook/ui/settings/SettingsFragment.kt (hunk 1 above)
Comment (zGararz) [reply app/src/main/java/com/example/itbook/ui/settings/SettingsFragment.kt]: nó thừa em xóa rồi ạ

File: app/src/main/java/com/example/itbook/utils/ActivityExtension.kt (hunk 3 above)
Comment (zGararz) [reply app/src/main/java/com/example/itbook/utils/ActivityExtension.kt:18]: sau khi chuyển ngôn ngữ thì phải restart lại thì nó mới áp dụng full, còn k thì cái bottom navigation nó k được chuyển ngôn ngữ ạ

**Pull request merged**